pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -92,10 +94,50 @@ func (c *APIClient) connect() error {
 	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
 	defer cancel()
 	version, err := c.client.Discovery().Version(ctx)
-	if err == nil {
-		log.Debugf("connected to kubernetes apiserver, version %s", version.GitVersion)
+	if err != nil {
+		return err
+	}
+	log.Debugf("connected to kubernetes apiserver, version %s", version.GitVersion)
+
+	resourcesAuthProbError := c.resourcesAuthProb()
+	if resourcesAuthProbError != nil {
+		return resourcesAuthProbError
+	}
+	log.Debug("Could successfully collect Pods, Nodes, Services and Events.")
+	return nil
+}
+
+func (c *APIClient) resourcesAuthProb() error {
```

Comment:
What do you think of something more explicit as method name, like:
`checkResourcesAuth`
As this method is quite hard to read, can you explain it ?